core/types/transaction_test.go

```diff
@@ -501,6 +502,10 @@ func encodeDecodeJSON(tx *Transaction) (*Transaction, error) {
 	if err := json.Unmarshal(data, &parsedTx); err != nil {
 		return nil, fmt.Errorf("json decoding failed: %v", err)
 	}
+
+	if !strings.Contains(string(data), "chainId") {
```

Comment:
Please remove. This is not the correct place for this check. These tests are checking round-trip serialization/deserialization of txs. You should make a test in https://github.com/ethereum/execution-apis for 1) the case where a legacy tx is not protected and therefore the json does not have `chainid` and 2) the case where it is protected and the json does have `chainid`.